### PR TITLE
added training checkpoint and metrics saving and resuming feature

### LINKED ISSUE #4: Adding General Checkpoint Saving and Loading for Training
Currently the training pipeline only saves the model params and will only allow loading back the model for inference/evaluation, currently we cannot reload the model params and continue training as optimizer variables / loss and other training metrics are not saved. Integrating the ability to save general training checkpoints that can be used to reload and continue training at the checkpoint could be very useful in the future. 

Info on how to do it here: https://pytorch.org/tutorials/beginner/saving_loading_models.html

### PR DESCRIPTION
**Pull request by jaycehaoranwang**
new:
- added general checkpoint saving and resuming training feature
- Plotting and saving of metrics in csv and generation of graphs even from resumed training
- added Mean Average Error (MAE) metric to track while training

### CODE REVIEW

**Comment by jaycehaoranwang** [comment]
realized I should save every epoch instead of at end of training
